packages/components/src/core/InputDropdown/index.tsx

```diff
@@ -51,53 +53,101 @@ const InputDropdown = (props: InputDropdownProps): JSX.Element => {
           className="styled-label"
           details={details}
           counter={counter}
+          sdsType={sdsType}
         >
           {renderLabelText({
             counter,
             details,
             isMinimal,
             label,
+            multiple,
+            sdsType,
             shouldPutAColonAfterLabel,
+            value,
           })}
         </StyledLabel>
-        {shouldRenderDetails && <StyledDetail>{details}</StyledDetail>}
+        {shouldRenderDetails && (
+          <StyledDetail>
+            {renderDetailsText({ details, sdsType, value })}
+          </StyledDetail>
+        )}
+        {shouldRenderInlineMinimalDetails && (
+          <StyledDetail>
+            {renderDetailsText({ details, sdsType, value })}
+          </StyledDetail>
+        )}
         {shouldRenderCounter && <StyledCounter>{counter}</StyledCounter>}
         <IconWrapper>
           <Icon sdsIcon="chevronDown" sdsSize="s" sdsType="interactive" />
         </IconWrapper>
       </LabelWrapper>
 
-      {isMinimal && (
+      {shouldRenderMinimalDetails && (
         <MinimalDetails
           shouldTruncateMinimalDetails={shouldTruncateMinimalDetails}
         >
-          {details}
+          {renderDetailsText({ details, sdsType, value })}
         </MinimalDetails>
       )}
     </StyledInputDropdown>
   );
 };
 
+interface RenderLabelTextProps {
+  counter: InputDropdownProps["counter"];
+  details: InputDropdownProps["details"];
+  isMinimal: boolean;
+  label: InputDropdownProps["label"];
+  multiple: InputDropdownProps["multiple"];
+  shouldPutAColonAfterLabel: InputDropdownProps["shouldPutAColonAfterLabel"];
+  sdsType: InputDropdownProps["sdsType"];
+  value: InputDropdownProps["value"];
+}
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
The following suggestion is a way to reduce complexity!

```ts
function renderLabelText(props: RenderLabelTextProps) {
  const { sdsType } = props;

  if (sdsType === "label") {
    return renderLabelTypeLabelText(props);
  }

  if (sdsType === "value") {
    return renderValueTypeLabelText(props);
  }

  return null;
}

function renderLabelTypeLabelText({
  isMinimal,
  label,
  counter,
  value,
  shouldPutAColonAfterLabel,
}: RenderLabelTextProps) {
  if (isMinimal) return label;

  return (counter || value) && shouldPutAColonAfterLabel ? `${label}:` : label;
}

function renderValueTypeLabelText({
  value,
  multiple,
  isMinimal,
  label,
  counter,
  details,
  shouldPutAColonAfterLabel,
}: RenderLabelTextProps) {
  if (!value || multiple) {
    if (isMinimal) return label;

    return counter && shouldPutAColonAfterLabel ? `${label}:` : label;
  }

  if (isMinimal) return value;

  return (counter || details) && shouldPutAColonAfterLabel
    ? `${value}:`
    : value;
}
```

Reply:
Oh woow, Thank you very much @tihuan 
It is much better this way 😯🤩